### PR TITLE
CSS Custom Properties for $spacing

### PR DESCRIPTION
**Pull request by richtabor**
### Description
This PR replaces the existing hard-coded utility values with new CSS Custom Properties for which themes may use to apply their spacing logic better to our blocks.

The `$spacing` map still exists to keep fallback values easily accessible, but for a theme to override, it'd need to simply provide values for the following: 

```
:root {
	--coblocks-spacing--0: 0;
	--coblocks-spacing--1: #{map-get($spacing, 1)};
	--coblocks-spacing--2: #{map-get($spacing, 2)};
	--coblocks-spacing--3: #{map-get($spacing, 3)};
	--coblocks-spacing--4: #{map-get($spacing, 4)};
	--coblocks-spacing--5: #{map-get($spacing, 5)};
	--coblocks-spacing--6: #{map-get($spacing, 6)};
	--coblocks-spacing--7: #{map-get($spacing, 7)};
	--coblocks-spacing--8: #{map-get($spacing, 8)};
}
```

### Types of changes
New feature (non-breaking change which adds functionality)

### Notes
We can review/test away, but lets hold merge for `2.0`. 

## Screenshots
This is how they're applied in action:
Screen Shot 2020-05-01 at 3 46 24 PM

### CODE REVIEW

File: src/blocks/post-carousel/styles/slick.scss [hunk 1]
```diff
@@ -13,7 +13,8 @@
 
 .slick-list {
 	display: block;
-	margin: 0 map-get($spacing, -2);
+	margin: 0;
+	margin-bottom: calc(var(--coblocks-spacing--2, #{map-get($spacing, 2)}) * -1);
```
[the remark this answers is not in the document]
Comment (AnthonyLedesma) [reply src/blocks/post-carousel/styles/slick.scss:17]: Oh there are two values next to the margin. Ignore this comment. 🥇

File: src/blocks/click-to-tweet/styles/style.scss [hunk 2]
```diff
@@ -50,7 +50,7 @@
 		word-break: break-all;
 
 		&:not(.has-button-color) {
-			background: $color-twitter;
+			background: var(--coblocks--color--twitter, #1da1f2);
```
[the remark this answers is not in the document]
Comment (richtabor) [reply src/blocks/click-to-tweet/styles/style.scss:53]: Yes intentional. If there’s no variable added it falls back to the hex value.

**Comment by AnthonyLedesma** [comment]
There are some really weird things going on with the tests. File uploads are failing in FireFox both local and in ci. I will need to revisit this Monday.

**Comment by AnthonyLedesma** [comment]
Cypress lacks support for sending native events https://github.com/cypress-io/cypress/issues/170#issuecomment-226518599 thus utilizing drag-and-drop features of the block editor has been problematic across browsers and as the editor changes between software releases. This change is an attempt to provide an agnostic method to upload files by exposing the raw input elements as targets. 

Created helper function for upload images and updated all related tests with exception to the Author block. The Author block requires changes that are outside of the scope of this PR. An issue has been raised [here](https://github.com/godaddy-wordpress/coblocks/issues/1495) with more detail.

Also included in this PR is a `.prettierrc.js` file which references the defaults from wp-scripts. This will accurately reflect the existing docs [here](https://github.com/godaddy-wordpress/coblocks/blob/02f4b2cfbfe0a05392fa01c6c43340817c774674/docs/contributors/getting-started.md#prettier) and will allow for out of box support for prettier code formatting in VS code IDE. 

A number of tests have changes to selector specificity which enables tests to pass across browsers consistently.

**Comment by AnthonyLedesma** [comment]
I will re-visit this issue and get the conflicts merged. I also wanted to undo some of the changes from this PR in favor of the changes from #1516

**Pull request merged**